File: java/src/main/java/com/svix/agentic_commerce_protocol/WebhookOrderStatus.java
// This file is @generated
package com.svix.agentic_commerce_protocol;

import com.fasterxml.jackson.annotation.JsonValue;
import com.svix.agentic_commerce_protocol.Utils.ToQueryParam;

public enum WebhookOrderStatus implements ToQueryParam {
    CREATED("created"),
    MANUAL_REVIEW("manual_review"),
    CONFIRMED("confirmed"),
    CANCELED("canceled"),
    SHIPPED("shipped"),
    FULFILLED("fulfilled");
    private final String value;

    WebhookOrderStatus(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: kotlin/src/main/kotlin/com/svix/agentic_commerce_protocol/WebhookOrderStatus.kt
// This file is @generated
package com.svix.agentic_commerce_protocol

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.encodeToJsonElement
import kotlinx.serialization.json.jsonPrimitive

@Serializable
enum class WebhookOrderStatus : ToQueryParam {
    @SerialName("created") CREATED,
    @SerialName("manual_review") MANUAL_REVIEW,
    @SerialName("confirmed") CONFIRMED,
    @SerialName("canceled") CANCELED,
    @SerialName("shipped") SHIPPED,
    @SerialName("fulfilled") FULFILLED;

    override fun toQueryParam() = Json.encodeToJsonElement(this).jsonPrimitive.content
}
